lambdas/tests/unit/handlers/test_bulk_upload_metadata_handler.py

```diff
@@ -0,0 +1,166 @@
+
+    expected_calls = [
+        call(
+            queue_url=MOCK_LG_METADATA_SQS_QUEUE,
+            message_body=EXPECTED_SQS_MSG_FOR_PATIENT_1234567890,
+            nhs_number="1234567890",
+        ),
+        call(
+            queue_url=MOCK_LG_METADATA_SQS_QUEUE,
+            message_body=EXPECTED_SQS_MSG_FOR_PATIENT_1234567891,
+            nhs_number="1234567891",
+        ),
+    ]
+    mock_sqs_service.send_message_with_nhs_number_attr.assert_has_calls(expected_calls)
+
+
+def test_handler_log_error_when_fail_to_get_metadata_csv_from_s3(
+    set_env, mock_s3_service, mock_sqs_service, caplog
+):
+    mock_s3_service.download_file.side_effect = ClientError(
+        {"Error": {"Code": "500", "Message": "file not exist in bucket"}},
+        "s3_get_object",
+    )
+    expected_err_msg = "An error occurred (500) when calling the s3_get_object operation: file not exist in bucket"
+
+    lambda_handler(None, None)
+
+    assert caplog.records[-1].message == expected_err_msg
+    assert caplog.records[-1].levelname == "ERROR"
+
+    mock_sqs_service.send_message_with_nhs_number_attr.assert_not_called()
+
+
+def test_handler_log_error_when_metadata_csv_is_invalid(
+    set_env, mocker, mock_sqs_service, caplog
+):
+    mocker.patch(
+        "handlers.bulk_upload_metadata_handler.download_metadata_from_s3",
+        return_value=MOCK_INVALID_METADATA_CSV,
+    )
+
+    lambda_handler(None, None)
+
+    assert "validation errors for MetadataFile" in caplog.records[-1].message
+    assert caplog.records[-1].levelname == "ERROR"
+
+    mock_sqs_service.send_message_with_nhs_number_attr.assert_not_called()
+
+
+def test_handler_log_error_when_failed_to_send_message_to_sqs(
+    set_env, mock_s3_service, mock_sqs_service, mock_tempfile, caplog
+):
+    mock_sqs_service.send_message_with_nhs_number_attr.side_effect = ClientError(
+        {
+            "Error": {
+                "Code": "AWS.SimpleQueueService.NonExistentQueue",
+                "Message": "The specified queue does not exist",
+            }
+        },
+        "SendMessage",
+    )
+    expected_err_msg = (
+        "An error occurred (AWS.SimpleQueueService.NonExistentQueue) when calling the SendMessage operation:"
+        " The specified queue does not exist"
+    )
+
+    lambda_handler(None, None)
+
+    assert caplog.records[-1].message == expected_err_msg
+    assert caplog.records[-1].levelname == "ERROR"
+
+
+def test_download_metadata_from_s3(mock_s3_service, mock_tempfile):
+    actual = download_metadata_from_s3(
+        staging_bucket_name=MOCK_LG_STAGING_STORE_BUCKET,
+        metadata_filename=METADATA_FILENAME,
+    )
+    expected = MOCK_METADATA_CSV
+
+    mock_s3_service.download_file.assert_called_with(
+        s3_bucket_name=MOCK_LG_STAGING_STORE_BUCKET,
+        file_key=METADATA_FILENAME,
+        download_path=f"{MOCK_TEMP_FOLDER}/{METADATA_FILENAME}",
+    )
+    assert actual == expected
+
+
+def test_csv_to_staging_metadata():
```

Comment:
Only the happy path is tested for this function. I'd like to see a test for what would happen if it gets some unexpected data plugged in, and see it raises the expected error

Reply:
Thanks. Added test for unhappy path (when the data in csv file is invalid)